src/test/storage/iterables_test.cpp

```diff
@@ -9,22 +10,26 @@
 #include "gtest/gtest.h"
 
 #include "storage/chunk_encoder.hpp"
+#include "storage/create_iterable_from_segment.hpp"
 #include "storage/dictionary_segment.hpp"
 #include "storage/dictionary_segment/dictionary_segment_iterable.hpp"
+#include "storage/encoding_type.hpp"
 #include "storage/fixed_string_dictionary_segment.hpp"
+#include "storage/lz4_segment/lz4_segment_iterable.hpp"
 #include "storage/reference_segment/reference_segment_iterable.hpp"
 #include "storage/table.hpp"
 #include "storage/value_segment.hpp"
 #include "storage/value_segment/value_segment_iterable.hpp"
 
 namespace opossum {
 
+template <typename DataType>
 struct SumUpWithIterator {
   template <typename Iterator>
   void operator()(Iterator begin, Iterator end) const {
     auto distance = end - begin;
 
-    _sum = 0u;
+    _sum = 0;
```

Comment:
Feel free to get rid of some whitespaces here if you like.. Too much for my taste.

Reply:
Also, in sumUp _sum is not initialized explicitly, here it is. I'd vote to initialize it in the struct directly `int sum{0}`.

Reply:
i guess the initial idea why sum was initialized outside of the struct was to have a SumUp method that can be called on multiple segments subsequently using the same sum variable. we would lose this functionality if we would initioalize it in the struct. i got rid of the multiple `= 0`s by initializing it outside of the struct. open for discussions.

Reply:
I have no opinion here. If it is reused, keep it that way. But I would still unify the structs (even if one of them is not reused).